src/test/java/org/padaiyal/utilities/commandline/CommandLineUtilityTest.java

```diff
@@ -0,0 +1,289 @@
+    nonTerminatingCommand
+        .setCommand(TypeOfCommandLine.POWERSHELL, "ping 127.0.0.1 -n 4294967295");
+    nonTerminatingCommand.setCommand(TypeOfCommandLine.ZSH, "cat");
+
+    timeoutDuration = Duration.ofSeconds(5);
+
+    operatingSystem = OperatingSystem.getOperatingSystem();
+
+    typeOfCommandLine = switch (operatingSystem) {
+      case WINDOWS -> TypeOfCommandLine.CMD;
+      case LINUX -> TypeOfCommandLine.BASH;
+      case MAC_OS_X -> TypeOfCommandLine.ZSH;
+      default -> null;
+    };
+  }
+
+  @Test
+  void testExecuteCommandString() throws IOException, InterruptedException,
+      TimeoutException, CommandLineNotFoundException {
+    String commandString = command.getCommand(typeOfCommandLine);
+    Response response = CommandLineUtility.executeCommand(commandString);
+    Assertions.assertEquals(0, response.getReturnCode());
+    Assertions.assertNotNull(response.getExecutionDuration());
+    Assertions.assertNotNull(response.getExecutionStartTimestamp());
+    Assertions.assertNotNull(response.getExecutionEndTimestamp());
+  }
+
+  @Test
+  void testExecuteCommandStringWithTimeout() throws IOException, InterruptedException,
+      TimeoutException, CommandLineNotFoundException {
+    String commandString = command.getCommand(typeOfCommandLine);
+    Response response = CommandLineUtility.executeCommand(commandString, timeoutDuration);
+    Assertions.assertEquals(0, response.getReturnCode());
+  }
+
+  @Test
+  void testExecuteNonTerminatingCommandStringWithTimeout()
+      throws IOException, InterruptedException, CommandLineNotFoundException {
+    boolean timeoutExceptionThrown = false;
+    String nonTerminatingCommandString = nonTerminatingCommand.getCommand(typeOfCommandLine);
+    try {
+      CommandLineUtility.executeCommand(nonTerminatingCommandString, timeoutDuration);
+    } catch (TimeoutException e) {
+      timeoutExceptionThrown = true;
+    }
+    Assertions.assertTrue(timeoutExceptionThrown);
+  }
+
+  @Test
+  void testExecuteCommand()
+      throws IOException, InterruptedException, TimeoutException, CommandLineNotFoundException {
+    Response response = CommandLineUtility.executeCommand(command);
+    Assertions.assertEquals(0, response.getReturnCode());
+  }
+
+  @Test
+  void testExecuteCommandWithTimeout() throws IOException, InterruptedException,
+      TimeoutException, CommandLineNotFoundException {
+    Response response = CommandLineUtility.executeCommand(command, timeoutDuration);
+    Assertions.assertEquals(0, response.getReturnCode());
+  }
+
+  @Test
+  void testExecuteNonTerminatingCommandWithTimeout() throws IOException, InterruptedException {
+    boolean timeoutExceptionThrown = false;
+    try {
+      CommandLineUtility.executeCommand(nonTerminatingCommand, timeoutDuration);
+    } catch (TimeoutException | CommandLineNotFoundException e) {
+      timeoutExceptionThrown = true;
+    }
+    Assertions.assertTrue(timeoutExceptionThrown);
```

Comment:
replace instances of these to Assertions.assertThrows()

Reply:
Done.